tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -6963,7 +6963,7 @@ QualType HLSLExternalSource::GetNthElementType(QualType type, unsigned index) {
     return (index == 0) ? type : QualType();
   case AR_TOBJ_COMPOUND: {
     // TODO: consider caching this value for perf
-    const RecordType *recordType = type->getAsStructureType();
+    const RecordType *recordType = type->getAsStructureOrClassType();
```

Comment:
We don't acknowledge the possibility of failure here (which is fine since we've already are under `AR_TOBJ_COMPOUND`, so this could be `getAsRecordType()`.